### DISCUSSION TITLE
Deleting device deletes the associated IP

### DISCUSSION THREAD

**Discussion by AnythingOverIP** [netbox-community/netbox#12043] [Q&A]
Not sure if it's by design, but if you delete a device that has a Primary IP assigned to it, the IP is deleted as well. Is there a way to only delete the device and remove the IP association instead of deleting it?

I have a colleague that deleted a bunch of Devices to recreate them with a different type and he has to recreate each IP. That has been a real waste of time :( 

Should this be treated as a bug? if so, I can open an issue.

**Comment by candlerb** [comment]
It's by design. You have to disassociate the IP addresses from the device/VM interfaces, before deleting the device, if you want to leave the IP addresses "floating" (unassigned).  You could write a small Custom Script to do this.

Aside: you can change the device type of an existing device, without deleting it and recreating it.  However, you may then need to rename the interfaces (and other components) to match the new device type if required. There is a [script](https://github.com/netbox-community/customizations/blob/master/scripts/add_device_type_components.py) which can create additional additional interfaces which are in the device type but not the device.

**Reply by AnythingOverIP** [reply]
I understand all this. My point is that the UI does not give any hint about what's about to happen:
![image](https://user-images.githubusercontent.com/25624251/227390691-5d1108ce-4081-4fd4-bc48-c32d91fe06a4.png)

At the very least, a warning should say that the corresponding IP will be deleted instead of being unassigned. Ideally, we should have an option to choose not to delete those... 

To me, it makes sense that a cable connected to the device being deleted is deleted as well, but the IP can still be used, along with corresponding info like DNS...

**Comment by a-belhadj** [comment]
Hello, 
Is it possible to have a more explicit warning in the delete message ?

**Comment by PieterL75** [comment]
Check 'issues' list, and search for 'delete'
You'll find some nice FR and discussions there.
